Apps/BMSQueryApp/app/src/main/java/uk/ac/cam/cares/jps/bmsqueryapp/view/tab/EditFragment.java

```diff
@@ -74,22 +76,62 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
+                    SingletonConnection.getInstance(this.getContext()).addToRequestQueue(jsonObjectRequest);
+                    hideKeyboardFrom(view.getContext(), view);
 //                binding.temperatureEdit.clearFocus();
-                hideKeyboardFrom(view.getContext(), view);
+                } else if (dataIRI.equals("https://www.theworldavatar.com/kg/ontobms/V_VAV_E-7-1_FlowSP_CARES")
+                        || dataIRI.equals("https://www.theworldavatar.com/kg/ontobms/V_VAV_E-7-2_FlowSP_CARES")
+                        || dataIRI.equals("https://www.theworldavatar.com/kg/ontobms/V_CAV_E-7-7_FlowSP_CARES")) {
+                    params.put("dataIRI", dataIRI);
+                    params.put("value", Double.parseDouble(editableAttributes.get(0).getValue()));
+                    params.put("clientProperties", "WRITE_CLIENT_PROPERTIES");
+                    JsonObjectRequest jsonObjectRequest = new JsonObjectRequest(Request.Method.POST, WACNET_WRITE_URL.build().toString(), params, response -> {
+                        try {
+                            String responseString = response.getString("message");
+                            Toast.makeText(this.getContext(), responseString, Toast.LENGTH_LONG).show();
+                        } catch (JSONException e) {
+                            throw new RuntimeException(e);
+                        }
+                    }, error -> Toast.makeText(this.getContext(), "Failed to submit the change, please resubmit later.", Toast.LENGTH_SHORT).show());
+                    SingletonConnection.getInstance(this.getContext()).addToRequestQueue(jsonObjectRequest);
+                    hideKeyboardFrom(view.getContext(), view);
+                    params = new JSONObject();
+                    //set control mode to CARES
+                    if (dataIRI.equals("https://www.theworldavatar.com/kg/ontobms/V_VAV_E-7-1_FlowSP_CARES")) {
+                        params.put("dataIRI", "https://www.theworldavatar.com/kg/ontobms/V_VAV_E_7-1_Control_Mode");
+                        params.put("value", 1.0);
+                        params.put("clientProperties", "WRITE_CLIENT_PROPERTIES");
+                    } else if (dataIRI.equals("https://www.theworldavatar.com/kg/ontobms/V_VAV_E-7-2_FlowSP_CARES")) {
+                        params.put("dataIRI", "https://www.theworldavatar.com/kg/ontobms/V_VAV_E_7-2_Control_Mode");
+                        params.put("value", 1.0);
+                        params.put("clientProperties", "WRITE_CLIENT_PROPERTIES");
+                    } else if (dataIRI.equals("https://www.theworldavatar.com/kg/ontobms/V_CAV_E-7-7_FlowSP_CARES")) {
+                        params.put("dataIRI", "https://www.theworldavatar.com/kg/ontobms/V_CAV_E_7-7_Control_Mode");
+                        params.put("value", 1.0);
+                        params.put("clientProperties", "WRITE_CLIENT_PROPERTIES");
+                    }
```

Comment:
Code duplication. May be better to create a dict for dataIRIToControlModeIRI and get the controlModelIRI with the dataIRI. Can consider to create separate file in `/uk/ac/cam/cares/jps/bmsqueryapp/data` for VAV and Canopyhood to store equipmentIRI-dataIRI-controlModeIRI, so all the info about these types are centralized in one place

Reply:
Same as above, code has been simplified with the addition of IRIMapping class that constructs a map containing the IRIs